### PR TITLE
Restore auto-add co-owner feature

### PR DESCRIPTION
**Pull request by xavierdecoster**
And select correct target account for package security policies.

### CODE REVIEW

**Comment by xavierdecoster** [comment]
@joelverhagen @skofman1 this is the commit you may want to zoom into: https://github.com/NuGet/NuGetGallery/pull/6289/commits/8ff0fcb27f6e771c6edd2cf9b40929c8e94a2b16

File: src/NuGetGallery/App_Start/DefaultDependenciesModule.cs [hunk 1]
```diff
@@ -395,6 +395,9 @@ protected override void Load(ContainerBuilder builder)
 
             RegisterCookieComplianceService(builder, configuration, diagnosticsService);
 
+            builder.RegisterType<MicrosoftTeamSubscription>()
+                .SingleInstance();
```
Comment (joelverhagen) [inline src/NuGetGallery/App_Start/DefaultDependenciesModule.cs]: Does this really need to be a singleton? Would prefer if it was a new instance each time unless we are concerned and have measure performance impact.

File: src/NuGetGallery/Security/MicrosoftTeamSubscription.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using System.Threading.Tasks;
+
+namespace NuGetGallery.Security
+{
+    public class MicrosoftTeamSubscription : IUserSecurityPolicySubscription
+    {
+        private Lazy<List<UserSecurityPolicy>> _policies = new Lazy<List<UserSecurityPolicy>>(InitializePoliciesList, isThreadSafe: true);
+
+        internal const string MicrosoftUsername = "Microsoft";
+        internal const string Name = nameof(MicrosoftTeamSubscription);
```
Comment (joelverhagen) [inline src/NuGetGallery/Security/MicrosoftTeamSubscription.cs]: This should be a const. Refactoring can break behavior here.

File: src/NuGetGallery/App_Start/DefaultDependenciesModule.cs (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGetGallery/App_Start/DefaultDependenciesModule.cs]: Changed to instance per lifetime-scope

File: src/NuGetGallery/Security/MicrosoftTeamSubscription.cs (hunk 2 above)
Comment (xavierdecoster) [reply src/NuGetGallery/Security/MicrosoftTeamSubscription.cs]: True. Updated.

**Pull request merged**